AngularApp/src/app/collapsible-menu/pipes/search.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import { Pipe, PipeTransform, Injectable } from '@angular/core';
+import { CollapsibleMenuItem } from '../components/collapsible-menu-item/collapsible-menu-item.component';
+
+@Pipe({
+    name: 'search',
+    pure: false
+})
+@Injectable()
+export class SearchPipe implements PipeTransform {
+    transform(items: CollapsibleMenuItem[], searchString: string) {
+        return searchString && items ? items.filter(item => item.label.toLowerCase().indexOf(searchString.toLowerCase()) >= 0) : items;
+    }
+}
+
+@Pipe({
+    name: 'searchmatch',
+    pure: false
+})
+export class SearchMatchPipe implements PipeTransform {
+    transform(label: string, searchString: string) {
+        if (searchString) {
+            let matchIndex = label.toLowerCase().indexOf(searchString.toLowerCase());
+            if (matchIndex >= 0) {
+                return `${label.substr(0, matchIndex)}<b>${label.substr(matchIndex, searchString.length)}</b>${label.substr(matchIndex + searchString.length)}`
```

Comment:
I would have to change how I did it because any style or class I pass on the <b> element will be stripped out as part of innerHtml. Figure this is a good enough start.